### PR TITLE
chore(profiling): fix crashtracker and ddup interfaces

### PR DESCRIPTION
**Pull request by sanchda**
This PR cleans up the interfaces for these two components
* Make sure the crashtracker can actually receive a runtime_id
* Updates tests

This doesn't actually change the behavior of any live code, so eliding the changelog for now.

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Comment by sanchda** [comment]
* **#9852** <a href="https://app.graphite.dev/github/pr/DataDog/dd-trace-py/9852?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @sanchda and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: ddtrace/internal/datadog/profiling/dd_wrapper/include/constants.hpp [hunk 1]
```diff
@@ -12,3 +13,12 @@ constexpr unsigned int g_backend_max_nframes = 512;
 
 // Maximum amount of time, in seconds, to wait for crashtracker send operations
 constexpr uint64_t g_crashtracker_timeout_secs = 5;
+
+// Name of the only runtime we support right now
+constexpr std::string_view g_runtime_name = "CPython";
```
Comment (brettlangdon) [inline ddtrace/internal/datadog/profiling/dd_wrapper/include/constants.hpp:18]: I think we do have some customers on pypy 🤔

File: tests/internal/crashtracker/test_crashtracker.py [hunk 2]
```diff
@@ -48,8 +48,8 @@ def test_crashtracker_config_bytes():
         crashtracker.set_url(b"http://localhost:1234")
         crashtracker.set_service(b"my_favorite_service")
         crashtracker.set_version(b"v0.0.0.0.0.0.1")
-        crashtracker.set_runtime(b"4kph")
         crashtracker.set_runtime_version(b"v3.1.4.1")
+        crashtracker.set_runtime_version(b"0")
```
Comment (brettlangdon) [inline tests/internal/crashtracker/test_crashtracker.py:52]:
```suggestion
        crashtracker.set_runtime_id(b"0")
```

?

File: ddtrace/internal/datadog/profiling/dd_wrapper/include/constants.hpp (hunk 1 above)
Comment (sanchda) [reply ddtrace/internal/datadog/profiling/dd_wrapper/include/constants.hpp:18]: Wait, really?  Do cpython dynamic libraries even load on pypy?  If not, then crashtracker won't even begin to work there.

Comment (brettlangdon) [reply ddtrace/internal/datadog/profiling/dd_wrapper/include/constants.hpp:18]: oh, CPython extensions, probably not, Cython ones should though... so.... 🤷🏻 

also, TIL: https://pyo3.rs/v0.9.2/pypy.html

Comment (sanchda) [reply ddtrace/internal/datadog/profiling/dd_wrapper/include/constants.hpp:18]: Ah--I'll just do the right thing and propagate the interface for these things.

Comment (brettlangdon) [reply ddtrace/internal/datadog/profiling/dd_wrapper/include/constants.hpp:19]: `platform.python_implementation()` / `platform.python_version()` / etc are probably the best to use... just to be safe

File: ddtrace/internal/datadog/profiling/ddup/_ddup.pyx [hunk 3]
```diff
@@ -1,12 +1,12 @@
 # distutils: language = c++
 # cython: language_level=3
 
-import platform
 from typing import Dict
 from typing import Optional
 from typing import Union
 
 import ddtrace
+import platform
```
[the remark this answers is not in the document]
Comment (sanchda) [reply ddtrace/internal/datadog/profiling/ddup/_ddup.pyx:9]: Yup--I messed up my branch, so I wanted to push everything ASAP before even running tests.  I'm still getting used to graphite.  😵 

Probably many other linting errors too.

File: ddtrace/internal/datadog/profiling/crashtracker/_crashtracker.pyx [hunk 4]
```diff
@@ -132,6 +139,8 @@ def start() -> bool:
     exe_dir = os.path.dirname(__file__)
     crashtracker_path = os.path.join(exe_dir, "crashtracker_exe")
     crashtracker_path_bytes = ensure_binary_or_empty(crashtracker_path)
+    set_runtime(platform.python_implementation())
+    set_runtime_version(platform.python_version())
```
[the remark this answers is not in the document]
Comment (sanchda) [reply ddtrace/internal/datadog/profiling/crashtracker/_crashtracker.pyx:143]: We can definitely go either way.  So
1. Don't do anything and just populate configs in the caller
2. Remove `platform` etc from here (think of it as a "pure" interface), expose the runtime_version and runtime_id stuff, then create something like ddtrace/internal/core/crashtracker.py where those configs are injected
3. Just skip the extra steps and put all the config right here, right now.

What would harmonize best with the idea of starting crashtracker ASAP in ddtrace startup?

[the remark this answers is not in the document]
Comment (brettlangdon) [reply ddtrace/internal/datadog/profiling/crashtracker/_crashtracker.pyx:143]: I like the idea of having a single "start" interface which just does the right config/setup/starting that is needed, but that doesn't have to live in this file.

I did want to have crashtracker live in `ddtrace/internal/core`... maybe I can put a `crashtracker.start()` interface there which does all the config and starting and etc, that way when we move crashtracker away from the profiling interface, then we can keep the same imports/API ?

[the remark this answers is not in the document]
Comment (sanchda) [reply ddtrace/internal/datadog/profiling/crashtracker/_crashtracker.pyx]: SGTM--dropped config from `_crashtracker.pyx` and exposed the runtime/runtime_version interfaces

**Review by taegyunkim — APPROVED** [review]

**Pull request merged**